File: program/find-roots-of-quadratic-equation/find_roots_of_quadratic_equation.rs
// Rust program to get the roots of a quadratic equation

fn roots(a:f32, b:f32, c:f32) {
    
    let disc:f32 = b * b - 4.0 * a * c;
    let root_a:f32 = (-b + disc.sqrt()) / (2.0 * a);
    let root_b:f32 = (-b - disc.sqrt()) / (2.0 * a);

    println!("Root1 = {}  ", root_a);
    println!("Root2 = {}  ", root_b);
}

fn main() {
    roots(1.0,-5.0,6.0);
}

File: program/program/find-roots-of-quadratic-equation/find_roots_of_quadratic_equation.hs
-- Haskell program to find roots of quadratic equation using the quadratic formula.

quadraticRoots :: Integral a => a -> a -> a -> (a, a)
quadraticRoots a b c
    | d >= 0    = (div (e + isqrt d) (2 * a), div (e - isqrt d) (2 * a)) -- Real roots
    | otherwise = (-1, -1)                                             -- Complex roots
    where
        d = b * b - 4 * a * c
        e = -b
        isqrt = floor . sqrt . fromIntegral

main :: IO ()
main = do
    let (root1, root2) = quadraticRoots 1 2 1
    putStrLn $ "Roots: " ++ show [root1, root2]
